src/model.rs

```diff
@@ -777,6 +777,39 @@ impl Collections {
         }
         self.vehicle_journeys = CollectionWithId::new(vehicle_journeys).unwrap();
     }
+
+    /// Many calendars are identicall and can be deduplicate
+    pub fn calendar_deduplication(&mut self) {
+        fn find_duplicate_calendar<'a>(
+            calendars: &'a [Calendar],
+            dates: &BTreeSet<NaiveDate>,
+        ) -> Option<&'a Calendar> {
+            for c in calendars {
+                if c.dates == *dates {
+                    return Some(c);
+                }
+            }
+            None
```

Comment:
This part could be written with the following syntax too.
```rust
calendars.iter().filter(|c| c.dates == *dates).next()
```
This is shorter and might allow to remove the `find_duplicate_calendar` function and put the code directly inline. If you prefer to keep the `find_duplicate_calendar`, then maybe the `for`-loop is more readable?

Reply:
`calendars.iter().find(|c| c.dates == *dates)`

Reply:
Ah yes, once again, you caught it. Maybe one day I'll think about it!